roles/insert_kernel_module/tasks/main.yml

```diff
@@ -1,39 +1,19 @@
 ---
-# This role loads one kernel_module specified in "kernel_module"
+# This role loads one kernel module specified in "insert_kernel_module_data"
 
-- fail:
+- name: "Fail on incorrect input : {{ insert_kernel_module_data.name }}"
+  ansible.builtin.fail:
     msg: >-
-      To run this role, you need to pass kernel_module object with attribute
+      To run this role, you need to pass insert_kernel_module_data object with attribute
       "name" defined.
-  when: kernel_module.name is not defined
+  when: insert_kernel_module_data.name is not defined
 
-# Detect what type of kernel module it is - builtin, loadable, or missing
-- import_tasks: detect_module.yml
+- name: "Import CPU info tasks"
+  ansible.builtin.import_tasks: cpu_info.yml
```

Comment:
I'm wondering whether it is worth it to separate this into an included file. Drawbacks: one need to take a look to a different file just for few tasks, and it's not clear from where insert_kernel_module_cpu_family and insert_kernel_module_cpu_model come from.

Reply:
The tasks were moved to main in 9202f63